pumpkin-lib/src/propagators/cumulative/time_table/time_table_over_interval_incremental.rs

```diff
@@ -81,105 +89,235 @@ impl<Var: IntegerVariable + 'static> TimeTableOverIntervalIncrementalPropagator<
         cumulative_options: CumulativePropagatorOptions,
     ) -> TimeTableOverIntervalIncrementalPropagator<Var> {
         let tasks = create_tasks(arg_tasks);
+        let parameters = CumulativeParameters::new(tasks, capacity, cumulative_options);
+        let dynamic_structures = DynamicStructures::new(&parameters);
+
         TimeTableOverIntervalIncrementalPropagator {
             time_table: Default::default(),
-            parameters: CumulativeParameters::new(tasks, capacity, cumulative_options),
-            time_table_outdated: false,
+            parameters,
+            dynamic_structures,
+            found_previous_conflict: false,
+            is_time_table_outdated: false,
+        }
+    }
+
+    /// Adds the added parts in the provided [`MandatoryPartAdjustments`] to the time-table; note
+    /// that all of the adjustments are applied even if a conflict is found.
+    fn add_to_time_table(
+        &mut self,
+        context: &PropagationContext,
+        mandatory_part_adjustments: &MandatoryPartAdjustments,
+        task: &Rc<Task<Var>>,
+    ) -> PropagationStatusCP {
+        let mut conflict = None;
+        // We consider both of the possible update ranges
+        // Note that the upper update range is first considered to avoid any issues with the
+        // indices when processing the other update range
+        for update_range in mandatory_part_adjustments.get_added_parts() {
+            // First we attempt to find overlapping profiles
+            match determine_profiles_to_update(&self.time_table, &update_range) {
```

Comment:
Is it important that the loop continues even after a conflict is found?

Reply:
Yes, since the removal of mandatory parts from other tasks could lead to the conflict which was previously found disappearing; this could be alleviated by first processing all of the removed parts and then processing all of the added mandatory parts but I believe this would lead to an increase in the overall overhead